aiobafi6/device.py

```diff
@@ -434,32 +452,26 @@ def _stop(self) -> None:
         elif self._connect_task is not None:
             self._connect_task.cancel()
         # Otherwise, if `_connect` is scheduled, cancel that and call
-        # `_sched_connect_or_signal_run_fut` directly because nothing else will.
+        # `_sched_connect_or_reset` directly because nothing else will.
         elif self._connect_timer is not None:
             self._connect_timer.cancel()
             self._connect_timer = None
-            self._sched_connect_or_signal_run_fut()
-
-    def _finish_run(self, _: asyncio.Future) -> None:
-        """Reset the run future to None.
-
-        This is the only completion callback for the run future and the only place where
-        it is reset to None, indicating that the device has fully stopped and could be
-        run again."""
-        _LOGGER.debug("%s: Stopped.", self.name)
-        self._run_fut = None
-        self._stop_requested = False
+            self._sched_connect_or_reset()
 
     # Availability
 
     @property
     def available(self) -> bool:
         """Return True when device is running and has values for critical properties."""
-        return self._available_event.is_set()
+        try:
```

Comment:
Untested, but you could likely simplify this to .done() and not .exception() to avoid the try

Reply:
No because the device is not available if the available future contains an exception.

Reply:
Right. I'm suggesting checking for the execution as well https://github.com/jfroy/aiobafi6/pull/72#discussion_r1303092305